### PR TITLE
Fixed credits for you buddy

### PR DESCRIPTION
**Pull request by ash47**
You should really consider crediting people for their work. It's one thing to steal someone's hard work, over 6 months of hard work has gone into Legends of Dota, and all we get is "THIS WAS 100% CREATED BY DARK SUPREMO WITH NO HELP FROM ANYONE!"

You should also update the workshop credits, and in reality, should add us as a contributor, since we made most of the addon.

### CODE REVIEW

**Comment by DafiniDesigns** [comment]
http://cdn.meme.am/instances/500x/60029622.jpg

**Comment by DarkSupremo** [comment]
The credit is already on  README.md, if you're blind, don't blame me.
But, as i already said on IRC, i'm rewriting the LUA almost from scratch, without your code, just waiting for time to finish it.

**Comment by ash47** [comment]
Perhaps you're blind. There is kV files, custom skills, custom units, loads of custom stuff aside from Lua you didn't credit, your credit under scopes what you stole. And 90% is just a random number you pulled out of thin air, the true number is much higher.

**Comment by DarkSupremo** [comment]
As i said above, i'm rewriting that all from scratch, even the KV's , should be finished next week.

**Comment by ash47** [comment]
So, merge the pull then update the credits at such a time when you have done what you claim you'll do.

**Comment by DarkSupremo** [comment]
I really don't know why you're that mad with it, both projects are free and open source, i created this mod just because you didn't added to your and i saw that many peoples (including me) liked to play this on d2ware times, and your name is already on credit with high %....


I'll just not merge because your comments on README.MD is basically offensive, edit it and i can merge..

Including:
- Basically all the good stuff in this game mode
- Basically nothing
- All of the code that wasn't written by Ash47 or BMD (didn't really understood the context of this...)

Others is fine for me...

**Comment by ash47** [comment]
How would you feel if you worked on a project for FREE for 6+ months, then someone comes in, takes your work, slaps their name on it, then claims to have made it themselves?

I'm all for learning from others code, I'm happy for others to play with legends of Dota and use it to make cool stuff, just don't claim you made it.


You will never know how it feels to have your work spoken because you won't ever have that kind of commitment required to make something new. I'm pissed because you are well known for stealing other people's work and slapping your name on it. A simple Google search will reveal this.

Once you've rewritten it, maybe I should download a copy, change the credits to say Ash47, then upload to the workshop?

If you insist on the edits, I'll make them tomorrow.

**Comment by DarkSupremo** [comment]
How i just claimed to made it myself alone if your credit is here?
I never, in any place said that i did this 100% by myself, even before adding your name to the credit (just took long time to add your name here because i was about 6 months without working on this project and i forgot, i started working on it again after the getdotastats...)

**Comment by ash47** [comment]
I see many places:

http://imgur.com/z5MYzDw

Number of people who visit that page? 30k+ ? Number of people who visit this repo? That's a private number but I bet it's tiny.

**Comment by DarkSupremo** [comment]
Ok, i really forgot about this, since i already said i was about 6 months without working on it... but i can add your credit here also without problem, it was not proposital, i just copied the description layout from another project because of the guide made by Wyk with How To Play.

I'm just updating the credit here based on your pull request, but removing the lines that i said above and adding few things that i did also.

**Comment by ash47** [comment]
I no longer care. It's my fault. I convinced myself that I worked on legends of Dota. I'll update it so it says you made it in the morning. Sorry to waste your time with my dillusions. I just wanted to feel like I was a good modder or something.

**Comment by DarkSupremo** [comment]
I learned a lot with your code, but that's only fun to me, an hobby, sorry if i forgot to add your credit.
I'll be glad also if you implement such mod on your LoD or create one yourself also, i didn't have many time to work on it also... my Job is as Programmer as well, with Java / C / PHP / Python, so i have not much free time to work on this, that's was why i used many of your code that was already done..

**Comment by ash47** [comment]
This is implanted in legends of Dota, but since you made lod, you already knew that.

The version in lod allows you to dynamically change between all the different x's, without 50 different addons.

We're just waiting for you to roll the update out, any eta?

**Comment by DarkSupremo** [comment]
I did that as an single addon in d2moddin, but i had problem with few skills that didn't worked as an custom abilities, such as: Drow Frost Arrow, Pudge Rot, and few more that i don't remember correctly.

Not sure if they fixed it on Source 2 already or not, i'm waiting for Spell Library that someone is working on...

Also there's another problem that i have to generate all skills/items description based on multiplier, it increase the size from about 3mb per language and multiplier avaliable...

The bigger problem is the free time to remake it also..

What update you mean?
Added the credit here already, i'll add it to workshop at night.

**Comment by ash47** [comment]
It could increase it by 20MB per language, it's irrelevant because of compression. It compresses to a tiny factor of the size when downloaded via steam, especially if you optimize the language file for compression.

**Comment by DarkSupremo** [comment]
I just wanted that instead of you saying bad about me on irc (as i saw on quote and someone told) you could actually talked to me before, i would had no problem in adding credit to you if i just forgot it...

But, do you have any plan in releasing it on LoD or separated addon?
Actually, since the way that lobby work (you can't know what game mode the host will choose) i don't actually like an addon with many game modes that change completly the game play.. some times i didn't want to play some game mode, but since i can't know it before the game started, that would be frustrating.

**Comment by DarkSupremo** [comment]
Many work that i had on this addon is balancing skills / items, to keep it fun to play to both sides and try make the game take more time to finish...

**Comment by MNoya** [comment]
img

Relax ash47 you're overreacting it, DarkSupremo never meant to "stole" from you and is being quite open and honest about giving you proper credit. You are known and renowned in the modding community by your awesome self-made job with LoD, easily the most polished Dota mod in the workshop, and no one will take that away from you. You're in all rights of requesting credits for your hardwork, but getting mad and undermining other people's addons that got popular using your code it's not the way to act about it, you're better than those MoD vultures.

**Comment by DarkSupremo** [comment]
Added credit to workshop.

**Comment by ash47** [comment]
@MNoya it worked, didn't it? I rest my case.

**Comment by DarkSupremo** [comment]
As i said before, you could just contacted me by issue here or email, without need of being offensive, it was not intentional, i just forgot to add the credits.

**Comment by ash47** [comment]
That would be too easy.

**Comment by paralin** [comment]
I should like to just say you guys might be upset about lack of attribution
but you have handled this situation fairly immaturely, from an outside
perspective...

**Comment by ash47** [comment]
Lol, immature is my middle name on #dota2mods

**Comment by DarkSupremo** [comment]
As i said before, i was already reworking this mod from scratch, since it was almost done, i got some time today to finish it.

I don't think that there's any code from you anymore, please check it and tell me if there's still some code "stolen" from you, if not i'll remove the credit from you, since i really didn't liked the way you handled that.

The code is already pushed to here, i'll do some more tests and improvements before releasing it to the public.

**Comment by paralin** [comment]
Understand ash the spirit of open source is that people can derive your work and use it to build things that others find useful. I don't know what kind of code license you released your source with, and if DarkSupremo is really violating something in that license **that you had at the time of releasing the parts he derived** you can persue him on those violations. But as it is, that's neither worth it nor probably even applicable. This is a small and tight knit community of people who share code with each other to create fun game modes as a hobby, not a business. It looks like you've received attribution on here, and Supremo is clearly dedicated to finishing his own unique version of the mod. Arguing over GitHub issues is not likely to get you anywhere fast.

**Comment by ash47** [comment]
I stopped arguing last night, lol >_>

So, how would you feel if I took d2moddin, changed the logos and put my name on it?

**Comment by DarkSupremo** [comment]
I kept you in the credits as: The base that was used to create the old version of this mod (before i reworked it from scratch), and i learned many things with LoD Source.

**Comment by paralin** [comment]
I released it open source under a license that requires people to give attribution and also to ask for explicit permission from me before use.

**Comment by ash47** [comment]
My issue wasn't really github, it was the workshop pages, obviously I saw my credit on here.

There was never a credit on the workshop version, prior to this happening.

**Comment by ash47** [comment]
So having no license listed means it's free to take credit for it?

**Comment by DarkSupremo** [comment]
Since there's no code "stolen" from you anymore, i think we can end this discussion.

**Comment by ash47** [comment]
Or is there? The plot thickens! Next time I won't be a dick about it. I learned something too, enjoy your day :)

**Comment by DarkSupremo** [comment]
I wish the same to you.

**Comment by ash47** [comment]
@DarkSupremo 

I don't understand how you can say you rewrote it from scratch. There are still huge chunks of LoD code in there, fair enough the variable names, but the comments? Why would you have word for word the comments I wrote, in my language, in your "from scrach" code?

To me, it just looks like you've renamed variables and then told me it is "from scratch"

https://github.com/DarkSupremo/Multiplier/blob/master/Workshop/base/game/dota_addons/power_multiplier/scripts/vscripts/addon_game_mode.lua#L57-L70

https://github.com/DarkSupremo/Multiplier/blob/master/Workshop/base/game/dota_addons/power_multiplier/scripts/vscripts/addon_game_mode.lua#L280-L364

You write your comments like this:

--no space after the dashes

I write my comments like this

-- A space after the dashes

A simple seach suggests at least 62 of the comments are not yours, if your code was written from scratch, it would have a consistent syntax.

Other obvious tells are that I use single quotes for my strings (this is some what rare, in my limited code reading experience from dota 2 mods), you are using double quotes for your strings.

I also noticed I write function calls like this(argumentWithNoSpace) and you write yours like this( argumentWithSpaces )

This time I will be clear: I am not mad you used my code, or learned from it. I'm mad you said you wrote it from scratch when you didn't.

As you put it yourself:

Since there's no code "stolen" from you anymore

Use the code, I don't care, just don't tell me you wrote it.

**Comment by DarkSupremo** [comment]
Really that you mentioned that i used the same variable name as you? for maxSkills and so?
Yeah, that Abaddon and Riki ulti is really from you, i missed that, since i'll add credit for you to that both, but the others thing can be a little based on something that had been done be you, but it's not the same, since the idea is to do the one thing, we have many ways to do that but all end up beeing similar.

**Comment by ash47** [comment]
This is not going to get out of hand like last night, please no more replies. Bye bye :)

**Comment by paralin** [comment]
Ash seriously you were the one who responded with another obsessive and
childish accusation, you can't act like you're not the one making things
get out of hand...

**Comment by ash47** [comment]
Well, he did say he used NONE of my code, that it was written FROM SCRATCH, I just wanted to show this was false :)

**Comment by DarkSupremo** [comment]
I also said to you check it, since i did that fast yesterday maybe i missed something.
I didn't throw everything alway and started in an new file, i removed everything that i was not using and rewrited what i was using, in an new way.

For example: instead of checking skills from dota 2 files, checking if it's valid or not banned, i'm using an pre-made KV file with only skills that i want to allow, and so on.

The comments and function call that you said about spacing, that's not valid, since that was not made in one day, and i still used the skeleton of method call and rewrited then, and so...

**Comment by ash47** [comment]
See, my issue is with the wording. You said from scratch:

from the very beginning, especially without making use of or relying on any previous work for assistance.

Which is simply not true. I have no issue with you using stuff from lod. I have an issue with you using it and telling me you didn't use it.

**Comment by DarkSupremo** [comment]
I NEVER said that i didn't used anything from you (even for assistence), ever after i said that i rewrited from scratch, i leaved you on credits as base (ideas) to make the current version.

Ok, i admit that i forgot to add the credits on workshop, but as i said before it was not intentional, since that was many months without changing that and i always increment the changelog and ctrl + c / ctrl + v to anothers versions i didn't saw that when i added the credits to here.

**Comment by ash47** [comment]
I'm talking about in this pull request, you said you no longer use ANY of the code from lod

Since there's no code "stolen" from you anymore

Perhaps stolen was the wrong word? I said stolen because you took ownership of it. The code that posed me right off was the multicast, I can remember sitting there for days writing that code, getting it to work, then you (whether it was by mistake or not) said you wrote it.

Sure, I offended you by writing "basically nothing", but my point wasn't invalid, and bringing attension to the issue (even if this is childish) has worked, you acknowledged that you didn't write certain parts of the code.

**Comment by DarkSupremo** [comment]
Actually, the multicast code was there but was not even used... it was not working because i didn't renamed it and nor created the ability _lod that it used... there's was many codes that was not being used and i was without free time to clean that up...

**Comment by ash47** [comment]
We are almost ok. The _lod multicast was in your branch.

To clarify: when you say "from scratch", this means you are removing or obfusicating code you didn't write yourself?

**Comment by DarkSupremo** [comment]
then it might be something else, the fact is that it was not working and i didn't even tried to make it work..

it mean that i removed many codes that was unnecessary and was not done by me, and rewrited most of codes that was written by you, but basically in an different way, like the example that i said above, to how i get skills now instead how it was with your code... but the idea and goal is the same..

**Comment by DarkSupremo** [comment]
i had problems with sometimes getting less skills than should and so, since it was not written by me, it was hard to debug and find the reason... that was what motived me rewriting it, knowing what i'm doing (before you even opened this pull request)

**Comment by paralin** [comment]
Ash you're still being a bit of a dick about something that ultimately
nobody cares about.

**Comment by ash47** [comment]
Paralin, I want to know how you when came about this pull request.

The other bit of context I forgot was that I was linked to this, and told "darksup said you helped him to code it, and said it was ok"

**Comment by DarkSupremo** [comment]
i never said that.

**Comment by DarkSupremo** [comment]
He probably received an email notification since he is still as Collaborator

**Comment by ash47** [comment]
That was what I was TOLD when someone pointed me to this add-on, I don't just sit on the workshop looking for people who use lod code XD
